### PR TITLE
Editor - Private Site Media

### PR DESCRIPTION
**Pull request by aforcier**
Fixes https://github.com/wordpress-mobile/WordPress-Editor-Android/issues/255, allowing display of images for private sites in the new editor and the editor's image settings dialog.

### CODE REVIEW

**Comment by aforcier** [comment]
The issue of occasionally missing calls to `WebViewClient.shouldInterceptRequest()` (https://github.com/wordpress-mobile/WordPress-Android/issues/3523) applies to this PR as well.

It rarely occurs on real devices (mostly emulators), so for now I'll make another issue in the editor repo to track this and hopefully find and address the problem in the future.

**Comment by aforcier** [comment]
Update: Moved the `setupUrlConnection()` utility method to the `utils` library so it can be reused in other parts of the app.

**Comment by aforcier** [comment]
Updated to use the utility method for adding auth headers added in #3548.

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorWebViewAbstract.java [hunk 1]
```diff
@@ -43,6 +58,64 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
...
+                    // Request and add an authorization header for HTTPS resource requests.
+                    // Use https:// when requesting the auth header, in case the resource is incorrectly using http://.
+                    // If an auth header is returned, force https:// for the actual HTTP request.
+                    String authHeader = mAuthHeaderRequestListener.onAuthHeaderRequested(UrlUtils.makeHttps(url));
+                    if (StringUtils.notNullStr(authHeader).length() > 0) {
+                        url = UrlUtils.makeHttps(url);
+                        headerMap.put("Authorization", authHeader);
+                    }
+
+                    HttpURLConnection conn = HTTPUtils.setupUrlConnection(url, headerMap);
```
Comment (maxme) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorWebViewAbstract.java]: This is not working when the url is "local" - when you try to add a local file for instance, we'll get a `FileURLConnection` and we don't want to call `setupUrlConnection` on it.

So we should check either check for the URL type (if the scheme is `http` or `https`), or  we should check the type (using `instanceOf`) returned by `new URL(url).openConnection()` in `HTTPUtils.setupUrlConnection()`

Comment (aforcier) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorWebViewAbstract.java]: Fixed! (87137f1)

**Comment by maxme** [comment]
Looks good! :shipit:

**Pull request merged**